store/src/main/java/com/dropbox/android/external/store4/Fetcher.kt

```diff
@@ -20,80 +23,95 @@ sealed class FetcherResult<T : Any> {
  * Note: Store does not catch exceptions thrown by a [Fetcher]. This is done in order to avoid
  * silently swallowing NPEs and such. Use [FetcherResult.Error] to communicate expected errors.
  *
- * See [nonFlowFetcher] for easily translating from a regular `suspend` function.
- * See [valueFetcher], [nonFlowValueFetcher] for easily translating to [FetcherResult] (and
+ * See [fromNonFlowFetcher] for easily translating from a regular `suspend` function.
+ * See [fromValueFetcher], [fromNonFlowValueFetcher] for easily translating to [FetcherResult] (and
  * automatically transforming exceptions into [FetcherResult.Error].
  */
-typealias Fetcher<Key, Output> = (key: Key) -> Flow<FetcherResult<Output>>
+interface Fetcher<Key, Output : Any> {
+    /**
+     * Returns a flow of the item represented by the given [key].
+     */
+    fun fetch(key: Key): Flow<FetcherResult<Output>>
 
-/**
- * "Creates" a [Fetcher] from a [flowFactory].
- *
- * Use when creating a [Store] that fetches objects in a multiple responses per request
- * network protocol (e.g Web Sockets).
- *
- * [Store] does not catch exception thrown in [flowFactory] or in the returned [Flow]. These
- * exception will be propagated to the caller.
- *
- * @param flowFactory a factory for a [Flow]ing source of network records.
- */
-fun <Key : Any, Output : Any> fetcher(
-    flowFactory: (Key) -> Flow<FetcherResult<Output>>
-): Fetcher<Key, Output> = flowFactory
+    operator fun invoke(key: Key) = fetch(key)
```

Comment:
1) why nor just `invoke`?
2) `fetch` is not a great name. This method does not do the fetch, rather, it returns a cold flow. I think calling it `fetch` is going to be a source of confusion around this point. maybe call it `flowFor`? It may also be good to change `operator invoke` to `operator get` for the same reason.

Reply:
just left invoke there for now, you are right fetch was not great.

i don't like operator get much since it feels like array access but it does something totally different. i think invoke is better.